tests/components/nibe_heatpump/test_climate.py

```diff
@@ -44,15 +56,249 @@ async def test_basic(
     coils[climate.current] = 20.5
     coils[climate.setpoint_heat] = 21.0
     coils[climate.setpoint_cool] = 30.0
-    coils[climate.mixing_valve_state] = "ON"
+    coils[climate.mixing_valve_state] = 20
     coils[climate.use_room_sensor] = "ON"
-    coils[unit.prio] = "HEAT"
+    coils[unit.prio] = "OFF"
     coils[unit.cooling_with_room_sensor] = "ON"
 
     await async_add_model(hass, model)
 
-    assert hass.states.get(entity_id) == snapshot(name="1. initial")
+    assert hass.states.get(entity_id) == snapshot(name="initial")
+
+    mock_connection.mock_coil_update(unit.prio, "COOLING")
+    assert hass.states.get(entity_id) == snapshot(name="cooling")
+
+    mock_connection.mock_coil_update(unit.prio, "HEAT")
+    assert hass.states.get(entity_id) == snapshot(name="heating")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 30)
+    assert hass.states.get(entity_id) == snapshot(name="idle (mixing valve)")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 20)
+    mock_connection.mock_coil_update(unit.cooling_with_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (only)")
+
+    mock_connection.mock_coil_update(climate.use_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (auto)")
+
+    mock_connection.mock_coil_update(unit.prio, None)
+    assert hass.states.get(entity_id) == snapshot(name="off (auto)")
+
+    if climate.active_accessory:
```

Comment:
Please avoid if statements in tests as it makes it harder to understand the test.

Reply:
I split this out to separate test